src/libs/ajax/AzureStorage.ts

```diff
@@ -111,6 +111,54 @@ export const AzureStorage = (signal?: AbortSignal) => ({
     }), notebooks)
   },
 
+  blobMetadata: (azureStorageUrl: string) => {
+    const getObjectMetadata = async () => {
+      const workspaceId = azureStorageUrl.split('/')[3].replace('sc-', '')
+      const fileName = _.last(azureStorageUrl?.split('/'))?.split('.')?.join('.')
+
+      try {
+        // assumption is made that container name guid in uri always matches the workspace Id guid it is present in
+        const { sas: { token } } = await AzureStorage(signal).details(workspaceId)
+
+        // instead of taking the url returned by azure storage, take it from the incoming url since there may be a folder path
+        const urlwithFolder = azureStorageUrl.substring(0, azureStorageUrl.lastIndexOf('/'))
+        const azureSasStorageUrl = _.flow(
+          _.split('?'),
+          _.head,
+          Utils.append(`/${fileName}?&${token}`),
+          _.join('')
+        )(urlwithFolder)
+
+        const res = await fetchOk(azureSasStorageUrl)
+        const headerDict = ParseHeaders(await res.headers.entries())
+
+        return { lastModified: headerDict['last-modified'], size: headerDict['content-length'], azureSasStorageUrl, workspaceId, fileName }
+      } catch (e) {
+        // check if file can just be fetched without sas token
+        try {
+          const res = await fetchOk(azureStorageUrl)
+          const headerDict = ParseHeaders(res.headers)
+          return { lastModified: headerDict['last-modified'], size: headerDict['content-length'], azureStorageUrl, workspaceId, fileName }
+        } catch (e) {}
+
+        throw e
+      }
+    }
+
+    const ParseHeaders = headerEntires => {
```

Comment:
Do we need to define a function for this? It looks like this is the same as [Object.fromEntries](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/fromEntries) or [_.fromPairs](https://lodash.com/docs/4.17.15#fromPairs).

Reply:
thank you! Object.fromEntries does the trick!